src/Attributes/CallbackType.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Tnapf\JsonMapper\Attributes;
+
+use Tnapf\JsonMapper\MapperInterface;
+
+abstract class CallbackType implements BaseType
```

Comment:
This is no longer a `CallbackType` since there's no callback. This looks more like `TypeTemplate` to me.

Is this really needed? `map` method can be a part of something like `MappableTypeInterface` that extends the `BaseType` interface.

Reply:
Definitely needed for something I'm working on currently

Reply:
> Definitely needed for something I'm working on currently

I mean, the this class can be safely replaced with an interface, because it's used as an interface.

Reply:
I see, main reason I'm using an abstract class is to make sure these properties are public readonly

Reply:
> I see, main reason I'm using an abstract class is to make sure these properties are public readonly

I see. It's just that not every implementation will require these properties, as you do in your projects that use this package. I would suggest a compromise of `MappableType` interface (extending the `BaseType` interface) and this class implementing that interface then. This is really optional, so if you don't feel like it then I see nothing wrong leaving it as it is.

Reply:
I would do this if it wasn't for the `isNullable` method. Which would have to be repeated in every CallbackType implementation. However, I think renaming `CallbackType` to `MappableType` probably a good decision as the name might be misleading